### PR TITLE
Fix interactive learning

### PR DESCRIPTION
**Pull request by dakshvar22**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/training/interactive.py [hunk 1]
```diff
@@ -966,10 +969,10 @@ async def _predict_till_next_listen(
     listen = False
     while not listen:
         result = await request_prediction(endpoint, conversation_id)
-        predictions = result.get("scores")
+        predictions = result.get("scores") or []
```
Comment (wochinge) [inline rasa/core/training/interactive.py:972]: how about raising an exception in case `predictions` is empty?

**Review by wochinge — APPROVED** [review]
Sorry for introducing this bug!

File: rasa/core/training/interactive.py (hunk 1 above)
Comment (dakshvar22) [reply rasa/core/training/interactive.py:972]: `RasaCoreException` or something more specific?

Comment (dakshvar22) [reply rasa/core/training/interactive.py:973]: Added `RasaCoreException`. Can we add any more specific information in the message, for example: Why such a situation may arise?

Comment (wochinge) [reply rasa/core/training/interactive.py:973]: Do we have something like an invalid config exception? I'd say that they need to retrain their model with a policy specified in the config then?

Comment (dakshvar22) [reply rasa/core/training/interactive.py:973]: Added.

**Review by wochinge — APPROVED** [review]
Thanks and sorry for breaking this 🙌🏻

**Pull request merged**